### ISSUE TITLE
Event listener of a created instance with stream doesn't work on firefox and edge browser

### ISSUE THREAD

**Issue by infocentre** [feross/simple-peer#143]
navigator.mediaDevices.getUserMedia({ audio: false, video: true }).then(function(stream) {
  var peer = new SimplePeer({ initiator: true, stream: stream })
  console.log(peer)
  peer.on('signal', function(data){
    console.log('printing.....')
  });
})

it is working well without stream:stream, and chrome. but on firefox it doesnt recognize event listener .on
so console.log doesn't  say 'printing....'  only not working on firefox, edge. on chrome it works well

**Comment by RationalCoding** [comment]
It looks like `self._pc.onnegotiationneeded` isn't being fired on Firefox when `initator: true`. I think this a Firefox regression although I can't find anything on it.

The hack used for wrtc works for this, so it could be applied to Firefox too? It doesn't seem to have any ill effects on Chrome either.
```javascript
    if (self._isWrtc) {
      self._pc.onnegotiationneeded()
    }
```

@infocentre Until there's a fix, you can manually fire that event like this:
```javascript
navigator.mediaDevices.getUserMedia({ audio: false, video: true }).then(function(stream) {
  var peer = new SimplePeer({ initiator: true, stream: stream })
  console.log(peer)
  peer.on('signal', function(data){
    console.log('printing.....')
  });
  peer._pc.onnegotiationneeded()
})
```

**Comment by feross** [comment]
We used to manually call `self._pc.onnegotiationneeded()` for Firefox, but I discovered it wasn't needed anymore since Firefox has finally implemented the event. Our mistake is that we're calling `addStream()` before we attach the 'negotiationneeded' handler, so it's firing before we're listening for it.

**Comment by feross** [comment]
Fixed in 6.4.1
